src/containers/explore/EntityDataContainer.js

```
@@ -0,0 +1,150 @@
+
+import { FQNS } from '../../core/edm/constants';
+import {
+  ENTITY_NEIGHBORS_MAP,
+  EXPLORE,
+  SELECTED_ENTITY_DATA
+} from '../../core/redux/constants';
+import { selectOrgDataSet, selectOrganization } from '../../core/redux/selectors';
+import { Routes } from '../../core/router';
+import { clipboardWriteText } from '../../utils';
+import { EntityDataGrid } from './components';
+
+const { getPropertyValue } = DataUtils;
+const { isPending } = ReduxUtils;
+const { isValidUUID } = ValidationUtils;
+
+const FlexContainer = styled.div`
+  display: flex;
+  justify-content: space-between;
+  margin-bottom: 30px;
+`;
+
+type Props = {
+  dataSetDataRoute ?:string;
+  dataSetId :UUID;
+  entityKeyId :UUID;
+  isModal ?:boolean;
+  organizationId :UUID;
+  organizationRoute ?:string;
+};
+
+const EntityDataContainer = ({
+  dataSetDataRoute,
+  dataSetId,
+  entityKeyId,
+  isModal = false,
+  organizationId,
+  organizationRoute,
+} :Props) => {
+  const dispatch = useDispatch();
+
+  const linkString = `https://openlattice.com/orgs/#${
+    Routes.ENTITY_DETAILS
+      .replace(Routes.DATA_SET_ID_PARAM, dataSetId)
+      .replace(Routes.ENTITY_KEY_ID_PARAM, entityKeyId)
+      .replace(Routes.ORG_ID_PARAM, organizationId)
+  }`;
+
+  const dataSet :Map<FQN, List> = useSelector(selectOrgDataSet(organizationId, dataSetId));
+  const entityData :?Map = useSelector((s) => s.getIn([EXPLORE, SELECTED_ENTITY_DATA], Map()));
+  const exploreEntityDataRS :?RequestState = useRequestState([EXPLORE, EXPLORE_ENTITY_DATA]);
+  const neighbors :?Map = useSelector((s) => s.getIn([EXPLORE, ENTITY_NEIGHBORS_MAP, entityKeyId], Map()));
+  const organization :?Organization = useSelector(selectOrganization(organizationId));
+
+  const name :string = getPropertyValue(dataSet, [FQNS.OL_DATA_SET_NAME, 0]);
+  const title :string = getPropertyValue(dataSet, [FQNS.OL_TITLE, 0]);
+
+  useEffect(() => {
+    if (isValidUUID(entityKeyId) && isValidUUID(dataSetId)) {
```

I'm thinking these might actually be more hurtful than helpful. If either is not a valid uuid, then nothing will happen, basically silent failure.  `exploreEntityData` calls `getEntityData`, which checks the id, so the saga will fail when it's invalid, and you can handle that error in the component. I think the same applies for `exploreEntityNeighbors`.